osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledTextBox.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        public MarginPadding Padding
+        {
+            get => base.Padding;
+            set
+            {
+                base.Padding = value;
+                base.Height = Height + base.Padding.Top;
+            }
+        }
+
+        public MarginPadding LabelPadding
+        {
+            get => label.Padding;
+            set => label.Padding = value;
+        }
+
+        public MarginPadding TextBoxPadding
+        {
+            get => textBox.Padding;
+            set => textBox.Padding = value;
+        }
+
+        public Color4 LabelTextColour
+        {
+            get => label.Colour;
+            set => label.Colour = value;
+        }
+
+        public Color4 BackgroundColour
+        {
+            get => content.Colour;
+            set => content.Colour = value;
+        }
+
+        public LabelledTextBox()
+        {
+            RelativeSizeAxes = Axes.X;
+            base.Height = default_height + Padding.Top;
+            CornerRadius = outer_corner_radius;
+            Masking = true;
+
+            InternalChild = new Container
+            {
+                RelativeSizeAxes = Axes.Both,
+                CornerRadius = outer_corner_radius,
+                Masking = true,
+                Children = new Drawable[]
+                {
+                    new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = OsuColour.FromHex("1c2125"),
+                    },
+                    new Container
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Height = default_height,
+                        Child = new GridContainer
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            Height = default_height,
+                            Content = new[]
+                            {
+                                new Drawable[]
+                                {
+                                    label = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.TopLeft,
+                                        Origin = Anchor.TopLeft,
+                                        Padding = new MarginPadding { Left = default_label_left_padding, Top = default_label_top_padding },
+                                        Colour = Color4.White,
+                                        TextSize = default_label_text_size,
+                                        Font = @"Exo2.0-Bold",
+                                    },
+                                    textBox = new OsuTextBox
+                                    {
+                                        Anchor = Anchor.TopLeft,
+                                        Origin = Anchor.TopLeft,
+                                        RelativeSizeAxes = Axes.X,
+                                        Height = default_height,
```

Comment:
Note that when you change this to `Axes.Both`, you're going to have to set `Height = 1` since `OsuTextBox` has a default height.